apps/src/aichat/views/ChatWorkspace.tsx

```diff
@@ -50,13 +65,35 @@ const ChatWorkspace: React.FunctionComponent<ChatWorkspaceProps> = ({
     }
   }, [messagesString, isWaitingForChatResponse]);
 
-  const dispatch = useAppDispatch();
+  const selectedStudentName = useMemo(() => {
+    if (viewAsUserId) {
+      const selectedStudent = Object.values(students).find(
+        student => student.id === viewAsUserId
+      );
+      if (selectedStudent) {
+        return getShortName(selectedStudent.name);
+      }
+    }
+    return null;
+  }, [viewAsUserId, students]);
 
-  const onCloseWarningModal = useCallback(
-    () => dispatch(setShowWarningModal(false)),
-    [dispatch]
+  // Teacher user is able to interact with chatbot.
+  const canChatWithModel = useMemo(
+    () =>
+      selectedTab !== WorkspaceTeacherViewTab.STUDENT_CHAT_HISTORY ||
+      !experiments.isEnabled(experiments.VIEW_CHAT_HISTORY),
+    [selectedTab]
   );
 
+  useEffect(() => {
+    // If we are viewing as a student, default to the student chat history tab if tab is not yet selected.
+    if (viewAsUserId && !selectedTab) {
+      setSelectedTab(WorkspaceTeacherViewTab.STUDENT_CHAT_HISTORY);
+    } else if (!viewAsUserId) {
+      setSelectedTab(null);
```

Comment:
minor - do we need to actually null out the tab when the viewAsUserId is undefined? We're not going to show the tabs anyway right, so does the value actually matter?

Also, if a teacher switches from teacher view to participant view and then back to teacher view, do we want to show them the tab they were previously on, or reset to student chat history? Looks like here we will reset back to student chat history but without this useEffect, we would stay on the tab they were on previously

Reply:
Yeah - I intentionally set this so that the student chat history would be the first thing they see vs. testing the student model. Lm ask product what they prefer. Thanks!

Reply:
Checked with @samantha-code and curriculum and they both prefer having student chat history viewed as default after switching back to teacher view.